### PR TITLE
8.0: Update What's New: SignalR stateful reconnect

### LINKED ISSUE #30675: .NET 8: What's New: SignalR: Stateful reconnect (Seamless reconnect) config option
Update the SignalR Seamless Reconnect section of [What's new in ASP.NET Core 8.0](https://learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-8.0?) to include the RC2 update that now uses `withStatefulReconnect` and is available now for Typescript Clients (.NET 8 RC2) as well as .NET clients using JavaScript (introduced in preview 5)

---
[Associated WorkItem - 171220](https://dev.azure.com/msft-skilling/Content/_workitems/edit/171220)

### PR DESCRIPTION
**Pull request by wadepickett**
Fixes #30675 

Updates the SignalR stateful reconnect section of the What's New for 8.0.

- SignalR seamless reconnect is renamed to _SignalR stateful reconnect_.
- Code example updated.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/release-notes/aspnetcore-8.0.md](https://github.com/dotnet/AspNetCore.Docs/blob/fe9ee39273a92a44a8c44ec0c622e690d9f7640b/aspnetcore/release-notes/aspnetcore-8.0.md) | [What's new in ASP.NET Core 8.0](https://review.learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-8.0?branch=pr-en-us-30804) |

### CODE REVIEW

**Comment by wadepickett** [comment]
I left the issue tracking link in for now at the bottom of the stateful reconnect section and will remove it in sync with the next release.

File: aspnetcore/release-notes/aspnetcore-8.0.md [hunk 1]
```diff
@@ -377,37 +377,37 @@ builder.On<string, string>("ReceiveMessage", (user, message) => ...
...
+Stateful reconnect achieves this by:
 
 * Temporarily buffering data on the server and client.
 * Acknowledging messages received (ACK-ing) by both the server and client.
 * Recognizing when a connection is returning and replaying messages that may have been sent while the connection was down.
 
 To opt-in to seamless reconnect:
 
-* Update the .NET client code to enable the `UseAcks` option:
+* Update the JavaScript or TypeScript client code to enable the `withStatefulReconnect` option:
```
Comment (BrennanConroy) [inline aspnetcore/release-notes/aspnetcore-8.0.md:392]: Should we also show the .NET client?

Comment (wadepickett) [reply aspnetcore/release-notes/aspnetcore-8.0.md:392]: Yikes!  Yes, sorry Brennan, I was focusing on JavaScript and TypeScript and left out the .NET Client.   I will add it back in!

File: aspnetcore/release-notes/aspnetcore-8.0.md [hunk 2]
```diff
@@ -377,37 +377,58 @@ builder.On<string, string>("ReceiveMessage", (user, message) => ...
...
+* Update the .NET client code to enable the `UseStatefulReconnect` option:
 
   ```csharp
   var hubConnection = new HubConnectionBuilder()
       .WithUrl("<hub url>",
                options =>
                {
-                  options.UseAcks = true;
+                  options.UseStatefulReconnect = true;
+                  options.StatefulReconnectBufferSize = 1000;  // Optional, defaults to 100,000
```
Comment (BrennanConroy) [inline aspnetcore/release-notes/aspnetcore-8.0.md:420]: I don't think this is how you enable it for the .NET client. I'll give a code snippet in a bit.

Comment (wadepickett) [reply aspnetcore/release-notes/aspnetcore-8.0.md:420]: @BrennanConroy I added the .NET Client example back in and updated based on what I was seeing in the SignalR code.  Could you verify if this is correct for the latest?

Comment (BrennanConroy) [reply aspnetcore/release-notes/aspnetcore-8.0.md:420]:
```csharp
var builder = new HubConnectionBuilder()
      .WithUrl("<hub url>")
      .WithStatefulReconnect();
builder.Services.Configure<HubConnectionOptions>(o => o.StatefulReconnectBufferSize = 1000);
var hubConnection = builder.Build();
```

Comment (wadepickett) [reply aspnetcore/release-notes/aspnetcore-8.0.md]: @BrennanConroy, I added in your .NET client sample.  I also made the hub config step consistent for both examples, so good to give that a look over.  Thanks again for the help.

File: aspnetcore/release-notes/aspnetcore-8.0.md [hunk 3]
```diff
@@ -377,37 +377,58 @@ builder.On<string, string>("ReceiveMessage", (user, message) => ...
...
   ```csharp
-  app.MapHub<AppHub>("/default", o => o.AllowAcks = true);
+  app.MapHub<MyHub>("/hubName", options =>
+  {
+      options.AllowStatefulReconnects = true;
+  });
   ```
 
-For more information on the progress of the seamless reconnect feature for ASP.NET Core 8.0, see [dotnet/aspnetcore #46691](https://github.com/dotnet/aspnetcore/issues/46691).
+For more information on the progress of the SignalR stateful reconnect feature for ASP.NET Core 8.0, see [dotnet/aspnetcore #46691](https://github.com/dotnet/aspnetcore/issues/46691).
```
Comment (BrennanConroy) [inline aspnetcore/release-notes/aspnetcore-8.0.md:431]: This should be removed now that 8.0 is done?

**Review by BrennanConroy — APPROVED** [review]

File: aspnetcore/release-notes/aspnetcore-8.0.md (hunk 3 above)
Comment (wadepickett) [reply aspnetcore/release-notes/aspnetcore-8.0.md]: I had purposely left it in until I had a doc update to point to instead more details, which I am working on next.  
I went ahead and removed it now.  I will add a link back in once I have the doc update in to point to.
Thanks.

**Pull request merged**